partner_communication/models/communication_attachment.py

```diff
@@ -56,7 +58,11 @@ def create(self, vals):
             })
             vals['attachment_id'] = attachment.id
 
+        vals['report_id'] = \
+            self.env['report']._get_report_from_name(vals['report_name']).id
```

Comment:
You should move this at the beggining of the method and only if `report_id` is not already defined in vals.
Then you should add the value of `report_id` inside the vals of the attachment created at line 53.